File: Spring boot (backend)/lms-backend/src/main/java/com/library/lmsbackend/service/StockService.java
package com.library.lmsbackend.service;

import com.library.lmsbackend.model.Stock;
import com.library.lmsbackend.repository.StockRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class StockService {

    @Autowired
    private StockRepository repository;

    public Stock saveStock(Stock stock) {
        return repository.save(stock);
    }

    public List<Stock> getAllStock() {
        return repository.findAll();
    }

    public Stock getStockById(Long id) {
        return repository.findById(id).orElse(null);
    }

    public void deleteStock(Long id) {
        repository.deleteById(id);
    }

    // PATCH: Update specific fields (e.g., just quantity)
    public Stock patchStock(Long id, Stock partialStock) {
        return repository.findById(id).map(stock -> {
            if (partialStock.getTitle() != null) stock.setTitle(partialStock.getTitle());
            if (partialStock.getAuthor() != null) stock.setAuthor(partialStock.getAuthor());
            if (partialStock.getSubject() != null) stock.setSubject(partialStock.getSubject());
            if (partialStock.getTotalQuantity() != null) stock.setTotalQuantity(partialStock.getTotalQuantity());
            if (partialStock.getAvailableQuantity() != null) stock.setAvailableQuantity(partialStock.getAvailableQuantity());
            if (partialStock.getStatus() != null) stock.setStatus(partialStock.getStatus());
            return repository.save(stock);
        }).orElse(null);
    }
}

File: React (frontend)/lms-backend_fe/src/api/stockService.js
import axiosInstance from "./axiosConfig";

export const createStock = async (payload) => {
  const { data } = await axiosInstance.post("/api/stock", payload);
  return data;
};

export const getAllStock = async () => {
  const { data } = await axiosInstance.get("/api/stock");
  return data;
};

export const getStockById = async (id) => {
  const { data } = await axiosInstance.get(`/api/stock/${id}`);
  return data;
};

export const patchStock = async (id, payload) => {
  const { data } = await axiosInstance.patch(`/api/stock/${id}`, payload);
  return data;
};

export const deleteStock = async (id) => {
  await axiosInstance.delete(`/api/stock/${id}`);
};
